agent/src/beerocks/slave/link_metrics/ieee802_3_link_metrics_collector.cpp

```diff
@@ -0,0 +1,279 @@
+    req.gen.rtgen_family = AF_PACKET; /*  no preferred AF, we will get *all* interfaces */
+
+    io.iov_base          = &req;
+    io.iov_len           = req.hdr.nlmsg_len;
+    rtnl_msg.msg_iov     = &io;
+    rtnl_msg.msg_iovlen  = 1;
+    rtnl_msg.msg_name    = &kernel;
+    rtnl_msg.msg_namelen = sizeof(kernel);
+
+    if (sendmsg(fd, (struct msghdr *)&rtnl_msg, 0) < 0) {
+        LOG(ERROR) << "Unable to send message through Netlink socket: " << strerror(errno);
+    } else {
+        int msg_done = 0; /* flag to end loop parsing */
+
+        /**
+         * Parse reply until message is done
+         */
+        while (!msg_done) {
+            int length;
+            struct nlmsghdr *msg_ptr; /* pointer to current message part */
+
+            struct msghdr rtnl_reply {
+            }; /* generic msghdr structure for use with recvmsg */
+
+            /* a large buffer to receive lots of link information */
+            char reply[IFLIST_REPLY_BUFFER];
+
+            io.iov_base            = reply;
+            io.iov_len             = IFLIST_REPLY_BUFFER;
+            rtnl_reply.msg_iov     = &io;
+            rtnl_reply.msg_iovlen  = 1;
+            rtnl_reply.msg_name    = &kernel;
+            rtnl_reply.msg_namelen = sizeof(kernel);
+
+            /**
+             * Read as much data as fits in the receive buffer
+             */
+            if ((length = recvmsg(fd, &rtnl_reply, 0)) != 0) {
+                for (msg_ptr = (struct nlmsghdr *)reply; NLMSG_OK(msg_ptr, length);
+                     msg_ptr = NLMSG_NEXT(msg_ptr, length)) {
+                    switch (msg_ptr->nlmsg_type) {
+                    case NLMSG_DONE:
+                        /**
+                         * This is the special meaning NLMSG_DONE message we asked for by using NLM_F_DUMP flag
+                         */
+                        msg_done = 1;
+                        break;
+                    case RTM_NEWLINK:
+                        /**
+                         * This is a RTM_NEWLINK message, which contains lots of information about a link
+                         */
+                        if (get_link_metrics(msg_ptr, local_interface_name, link_metrics)) {
+                            msg_done = 1;
+                            result   = true;
+                        }
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    return result;
+}
+
+/**
+ * @brief Gets link metrics of an Ethernet network interface.
+ *
+ * Gets link metrics for given Ethernet network by means of a Netlink socket using NETLINK_ROUTE
+ * protocol.
+ *
+ * @param[in] local_interface_name Name of the Ethernet network interface.
+ * @param[in, out] link_metris Link metrics structure with read values.
+ *
+ * @return True on success and false otherwise.
+ */
+static bool get_link_metrics(const std::string &local_interface_name, sLinkMetrics &link_metrics)
+{
+    bool result = false;
```

Comment:
Not needed, just return true of false where needed.

Reply:
Single vs. Multiple exit point, that's the question.
Have you agreed on which one to use or is it up to the programmer? I haven't seen anything about this topic in `CODINGSTYLE.md`

Reply:
I think in general we favour multiple exit points, it keeps the code indentation shorter and is what we have in most of prplmesh. I will update the `CODYINGSTULE.md` if agreed - @fvbogaert you have a chance for another poll :)

Reply:
We use multiple exit points everywhere. Because normally, all cleanup is automatic since we use smart pointers etc.